concent_api/gatekeeper/views.py

```diff
@@ -154,6 +158,30 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
     if len(transfer_token_paths_to_files) != len(set(transfer_token_paths_to_files)):
         return gatekeeper_access_denied_response('File paths in the token must be unique', path_to_file, loaded_golem_message.subtask_id, client_public_key)
 
+    if any((file["checksum"] is None or len(file["checksum"]) == 0 or file["checksum"].isspace()) for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The variable is empty', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+    if any(not isinstance(file["checksum"], str) for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The variable must be string', path_to_file, loaded_golem_message.subtask_id, client_public_key)
```

Comment:
This check should be above. Otherwise trying to call `isspace()` may result in an exception.

And if you check if it's `str` you don't have to specifically check for `None`.

Reply:
When I check `str(None)` , it return `'None'`, so it's work properly ?